junixsocket-common/src/main/java/org/newsclub/net/unix/AFSelector.java

```diff
@@ -110,47 +111,51 @@ public int select() throws IOException {
 
   @SuppressWarnings("PMD.CognitiveComplexity")
   private int select0(int timeout) throws IOException {
-    PollFd pfd;
     synchronized (this) {
```

Comment:
Why lock on `this` when we now also lock on `publicSelectedKeys`?

Reply:
That's a good question. In the standard lib, they lock on both for most of the implementation, but I think we can push down the lock for selected keys to leave `initPollFd` only locked under `this`.